nuxeo-drive-core/src/main/java/org/nuxeo/drive/listener/NuxeoDriveGroupUpdateListener.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Antoine Taillefer <ataillefer@nuxeo.com>
+ */
+package org.nuxeo.drive.listener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.drive.service.NuxeoDriveEvents;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.repository.RepositoryManager;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.EventContext;
+import org.nuxeo.ecm.core.event.EventService;
+import org.nuxeo.ecm.core.event.PostCommitFilteringEventListener;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+import org.nuxeo.ecm.platform.usermanager.UserManagerImpl;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Post-commit asynchronous listener that handles group change events fired by the {@link UserManager}.
+ * <p>
+ * For all the documents carrying an ACL impacted by a changed group or one of its ancestors it fires the
+ * {@link NuxeoDriveEvents#GROUP_UPDATED} event that is handled by the synchronous
+ * {@link NuxeoDriveFileSystemDeletionListener}.
+ *
+ * @since 9.2
+ */
+public class NuxeoDriveGroupUpdateListener implements PostCommitFilteringEventListener {
+
+    protected List<String> acceptedEvents = Arrays.asList(UserManagerImpl.GROUPCREATED_EVENT_ID,
+            UserManagerImpl.GROUPDELETED_EVENT_ID, UserManagerImpl.GROUPMODIFIED_EVENT_ID);
+
+    @Override
+    public boolean acceptEvent(Event event) {
+        return event.getContext() != null
+                & UserManagerImpl.USER_GROUP_CATEGORY.equals(
+                        event.getContext().getProperty(DocumentEventContext.CATEGORY_PROPERTY_KEY))
+                && acceptedEvents.contains(event.getName());
+    }
+
+    @Override
+    public void handleEvent(EventBundle events) {
+        for (Event event : events) {
+            EventContext context = event.getContext();
+            if (context == null) {
+                return;
+            }
+            String groupName = (String) context.getProperty(UserManagerImpl.ID_PROPERTY_KEY);
+            if (StringUtils.isBlank(groupName)) {
```

Comment:
In fact it can be user-provided if `UserManager#notifyGroupChanged(String groupName, String eventId, List<String> ancestorGroupNames)` has been called from an external source.